go/vt/vtgate/vindexes/vschema.go

```diff
@@ -216,11 +237,12 @@ func BuildKeyspaceSchema(input *vschemapb.Keyspace, keyspace string) (*KeyspaceS
 		},
 	}
 	vschema := &VSchema{
-		uniqueTables:   make(map[string]*Table),
+		globalTables:   make(map[string]*Table),
 		uniqueVindexes: make(map[string]Vindex),
 		Keyspaces:      make(map[string]*KeyspaceSchema),
 	}
 	buildKeyspaces(formal, vschema)
+	buildReferences(formal, vschema, true /*allowUnresolvedReferences*/)
```

Comment:
I think this is not needed on `BuildKeyspaceSchema` as it is used for validation for a schema provided for a keyspace and here we are anyways ignoring unresolved references.